libraries/chain/include/eosio/chain/abi_serializer.hpp

```diff
@@ -183,11 +183,24 @@ namespace impl {
       void check_deadline()const { yield( recursion_depth ); }
       abi_serializer::yield_function_t get_yield_function() { return yield; }
 
+      void start_action_serialization() {
+         if (max_action_serialization_time.count() > 0) {
+            fc::time_point deadline = fc::time_point::now().safe_add(max_action_serialization_time);
+            yield = [deadline, y=yield, max=max_action_serialization_time](size_t depth) {
```

Comment:
I think there is a potential problem here. If we call `start_action_serialization` twice on the same `abi_traverse_context`, the second time `yield` will be the already updated version of yield, so we are creating a cascade of lambda calls. Should we save the original value of `yield` in the constructor?

Reply:
Updated to call from constructor of a new context type.